vnext/Chakra/ChakraBytecodeStore.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ChakraBytecodeStore.h"
+#include "ChakraHelpers.h"
+#include "ChakraUtils.h"
+#include "Crypto.h"
+
+#pragma warning(push)
+#pragma warning(disable : 4244)
+#pragma warning(disable : 4245)
+#include <boost/crc.hpp>
+#pragma warning(pop)
+
+#include <boost/algorithm/hex.hpp>
+#include <cstdio>
+
+namespace facebook::react {
+
+///
+/// Metadata about a bytecode file that is persisted on disk
+///
+#pragma pack(push, 1)
+struct BytecodeFileHeader {
+  uint64_t fileFormatVersion;
+  HashedScript::Sha256Hash scriptHash;
+  ChakraVersionInfo chakraVersion;
+};
+#pragma pack(pop)
+
+constexpr uint32_t CurrentBytecodeFileFormatVersion = 3;
+static_assert(sizeof(BytecodeFileHeader) == 56, "CurrentBytecodeFileFormatVersion out of date");
+
+const wchar_t *BytecodeScratchSubdirectory = L"ChakraBytecode";
+const wchar_t *BytecodeFileExtension = L".bytecode";
+
+ChakraBytecodeStore::ChakraBytecodeStore(const std::filesystem::path &instanceScratchPath) {
+  m_bytecodeDirectory = instanceScratchPath / BytecodeScratchSubdirectory;
+}
+
+std::unique_ptr<JSBigString> ChakraBytecodeStore::tryObtainCachedBytecode(const HashedScript &script) {
+  auto bytecodePath = bytecodePathForScript(script);
+  auto bytecodeBodyStr = std::make_unique<FileMappingBigString>(bytecodePath.string(), sizeof(BytecodeFileHeader));
```

Comment:
>bytecodeBodyStr [](start = 7, length = 15)

(nit) please don't add type to the name, this is not needed here. bytecodeBody is enough, no?